### PR TITLE
[Lock] Make SemaphoreStore::isSupported() internal

### PR DESCRIPTION
**Pull request by maidmaid**
| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/pull/24053#discussion_r136391875
| License       | MIT
| Doc PR        | /

### CODE REVIEW

File: src/Symfony/Component/Lock/Store/SemaphoreStore.php [hunk 1]
```diff
@@ -33,6 +33,10 @@ class SemaphoreStore implements StoreInterface
      */
     public static function isSupported($blocking = null)
     {
+        if (null !== $blocking) {
+            @trigger_error(sprintf('The "$blocking" argument of the "%s" is deprecated since 3.4 and will be removed in 4.0.', __METHOD__), E_USER_DEPRECATED);
```
Comment (jderusse) [inline src/Symfony/Component/Lock/Store/SemaphoreStore.php]: Internal call should not trigger deprecation (this argument is used by component console)

File: src/Symfony/Component/Lock/CHANGELOG.md [hunk 2]
```diff
@@ -5,3 +5,4 @@ CHANGELOG
 -----
 
  * added the component
+ * deprecated the `$blocking` argument of the `SemaphoreStore::isSupported()` method
```
Comment (jderusse) [inline src/Symfony/Component/Lock/CHANGELOG.md]: The component is not released yet. Does it make sense to deprecate think here?

File: src/Symfony/Component/Lock/Store/SemaphoreStore.php (hunk 1 above)
Comment (maidmaid) [reply src/Symfony/Component/Lock/Store/SemaphoreStore.php]: But how know if call is internal or not?

Comment (chalasr) [reply src/Symfony/Component/Lock/Store/SemaphoreStore.php]: By passing `false` as second (extra, implicit) argument in each internal call, then triggering only if `1 === func_num_args() || func_get_arg(1)`. But wait, @jderusse do we really need to expose this argument in 3.4? Would it be enough to not add it and consider it exists only for internal calls?

File: src/Symfony/Component/Lock/CHANGELOG.md (hunk 2 above)
Comment (maidmaid) [reply src/Symfony/Component/Lock/CHANGELOG.md]: It's usefull to see deprecations linked to 3.4 here I think, else no textual informations is written elsewhere.

File: src/Symfony/Component/Lock/Store/SemaphoreStore.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Lock/Store/SemaphoreStore.php]: let's make it final and move on :)

Comment (chalasr) [reply src/Symfony/Component/Lock/Store/SemaphoreStore.php]: @nicolas-grekas doesn't https://github.com/symfony/symfony/pull/24053#discussion_r136384378 still apply when calling the method?

Comment (nicolas-grekas) [reply src/Symfony/Component/Lock/Store/SemaphoreStore.php]: nothing will break, so we don't care

Comment (chalasr) [reply src/Symfony/Component/Lock/Store/SemaphoreStore.php]: ok then :)

File: src/Symfony/Component/Lock/CHANGELOG.md (hunk 2 above)
Comment (fabpot) [reply src/Symfony/Component/Lock/CHANGELOG.md]: Lock was not part of 3.3, so there is nothing to deprecate.

File: src/Symfony/Component/Lock/Store/SemaphoreStore.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Component/Lock/Store/SemaphoreStore.php]: I would just mark this method as `@internal`

Comment (maidmaid) [reply src/Symfony/Component/Lock/Store/SemaphoreStore.php]: `@internal` used.

File: src/Symfony/Component/Lock/CHANGELOG.md (hunk 2 above)
Comment (maidmaid) [reply src/Symfony/Component/Lock/CHANGELOG.md]: Ok, line removed.

**Review by xabbuh — APPROVED** [review]

**Review by chalasr — APPROVED** [review]

**Review by javiereguiluz — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @maidmaid.

**Pull request merged**